### ISSUE TITLE
Water depth issue 2

### ISSUE THREAD

**Issue by jeremyEudaric** [csdms-contrib/fwdet#21]
Sorry about that and thank you for your reply and kindness l finally downloaded the DEM from MERIT DEM. l just downloaded the the DEM ( un burned ) and used your function with my polygon ( the CRS is the same) :

Unfortunately l still have a poor match the water depth does not take in count the all flood map (they are no error message)

![zz](https://github.com/csdms-contrib/fwdet/assets/79277563/ffa86a5a-8ff3-46ec-9058-2d4a4c93fc0c)

Looks like the DEM is not the issue because l used MERIT DEM. l am a bit lost to be honest do you have an idea ? :)

The link for the data https://drive.google.com/drive/folders/18km7boOaIIRpoDeJ7S1IlfVWItbRtolq?usp=drive_link

Thank a lot sorry for that

**Comment by jeremyEudaric** [comment]
![Screenshot from 2024-05-18 21-19-27](https://github.com/csdms-contrib/fwdet/assets/79277563/71b78732-82de-44a2-861a-5a7b85a83b44)
The water depth is between 0 and 215 m but its should be as your model between 0-13 m . May be we could organized a quick zoom call next that could be easier ? Thanks a lot :)

**Comment by cefect** [comment]
Sorry to hear it's still not working. Is the previous link still the most recent attempt? Having extremely large depths in a few cells is not surprising.

**Comment by jeremyEudaric** [comment]
Yes its the new one with the new data and DEM from MERIT DEM with this link : https://drive.google.com/drive/folders/18km7boOaIIRpoDeJ7S1IlfVWItbRtolq?usp=drive_link

Yes but l want to overlay the water depth with building footprints then if a building affected by a water depth of 215 m looks not realistic :/
![Screenshot from 2024-05-18 23-06-34](https://github.com/csdms-contrib/fwdet/assets/79277563/8440135c-301d-4aba-8f99-0defbe9f241a)

l tried with the data that you provided to test they are the same issue but less important : 

![Screenshot from 2024-05-18 23-09-38](https://github.com/csdms-contrib/fwdet/assets/79277563/8b143c0c-3226-4cc5-8514-3f760555aa40)

l really need a water depth precise to estimate the buildings affected.

Thank you for your help

**Comment by cefect** [comment]
I think the issue is that your inundation polygon is shifted west (relative to your DEM). As FwDET operates by extrapolating the elevations at the edges of the polygon, this causes it to extrapolate some very large depths.  
![image](https://github.com/csdms-contrib/fwdet/assets/47092034/e7164981-fdbf-4a0b-a53b-01ec35af1212)

Regardless, if you're looking to get water depths at buildings, I don't think you'll be very satisfied with a 25m DEM for FwDET.  I'd try and get something closer to 1m resolution.

Also, it looks like the northern edge of your inundation is some sort of administrative boundary? This will not work. The inundation polygon needs to reflect the actual flooded region (at least for one contiguous region).

![image](https://github.com/csdms-contrib/fwdet/assets/47092034/1e8bef80-8c43-4ea2-8bd0-e7ea444f0878)

**Comment by jeremyEudaric** [comment]
Good morning for your time and kindness the inundation polygon is portion of the flood event just to test and debug the code. the all flood polygon is on this link : https://drive.google.com/drive/folders/18km7boOaIIRpoDeJ7S1IlfVWItbRtolq?usp=drive_link
l think we are close to find a solution and thanks a lot for this. Do you have an idea how l could have a perfect match with my polygon and DEM ?
To give you the context l did as you explained in your paper used Sentinel-1 and l did a thresolding on a python code in order to get the flood map after l geo references manually the plot on QGIS ( l guess its not possible to do it automatically base on the DEM). Both the polygon and the DEM have the same extent. What do you think ?

Thank you for your help

**Comment by jeremyEudaric** [comment]
Do you know where l can find 1 m resolution for my DEM and if l can transform with QGIS my DEM from MERIT DEM  with 1 m resolution ? Sorry for all those questions l am trying to fix for 1 months , sorry for that :)

**Comment by cefect** [comment]
When testing on a portion of the data, its best to select a single contiguous flooded polygon... don't do any clipping.
![image](https://github.com/csdms-contrib/fwdet/assets/47092034/76d5854a-ef3c-48b1-a54d-d5de11af6d90)

To get a hydraulic match between the DEM and the polygon a more sophisticated hydraulic modelling solution is required; however, this means you're inundation footprint will not match your observations as well. FwDET (and tools like it) are designed to derive water depths more directly from inundations; however, if your observations are inconsistent with the terrain, such methods are not so useful. 

I know very little about computing inundation footprints from Sentinel data (and don't know which paper you are referring to) so I'm afraid I can not help with this.

For a higher resolution DEM, I suggest searching open data repositories for your region of interest (e.g., Europe and Greece) or asking local or regional experts. 

While you can increase the resolution of your data easily with tools like QGIS (i.e., gdal_warp), this is a simple data transformation and does not add any additional information and therefore will not be helpful for you. 

Regardless, I think your specific problem is related to some projection of the inundation polygon that caused the ~500m westward shift I referred to earlier. I suggest revisiting the workflow  you followed and paying close attention to your projections. Maybe switching to a UTM projection (e.g., EPSG: 32634) early in your workflow may help avoid this shift. 

![image](https://github.com/csdms-contrib/fwdet/assets/47092034/b9bc567e-b78c-423c-ac69-f5f00173a63f)

**Comment by jeremyEudaric** [comment]
Thanks for your help, do you who did the inundation map in the example online? l would like to contact this person. On this paper they are doing the buildings damage assessment with a DEM 30 m and 10 m resolution its why l was a bit confused : 

https://ieeexplore.ieee.org/document/9242297

![ppp](https://github.com/csdms-contrib/fwdet/assets/79277563/d425708a-89e6-4087-8314-c9cbb8db7a54)

**Comment by cefect** [comment]
The appropriate resolution is study specific. But finer resolution is always better for accuracy.

**Comment by jeremyEudaric** [comment]
Sorry but l am a bit confused because l am trying to reproduce exactly what you did with a water depth between 0 and 14 m as on your paper and  with the data provided. l am looking for a rough analyses  by overlaying the water depth map with the buildings as you did on the paper above with huttle Radar Topography Mission, 30-m spatial resolution and the USGS NED (10-m spatial resolution). 

Thank for your help l will let know once l fixed my issue

**Comment by jeremyEudaric** [comment]
Thank very much for your time an amazing support :)
